workitem/workitem_repository.go

```diff
@@ -104,6 +104,10 @@ func (r *GormWorkItemRepository) Save(ctx context.Context, wi app.WorkItem) (*ap
 	if res.Version != wi.Version {
 		return nil, errors.NewVersionConflictError("version conflict")
 	}
+	// Changing the type of a work item is prohibited.
+	if res.Type != wi.Type {
+		return nil, errors.NewBadParameterError("type", wi.Type).Expected(res.Type)
```

Comment:
Does this make it fail if the BaseType is included in the REST Api Patch? REST Api should apply what it can and ignore what it can't.